[lib] Don't use -rpath

chrissie-c opened the pull request:
rpath seems to annoy some analysis tools. We
use it to locate the crypto & compression plugins but libknet also needs
to use the plugins from the build-tree when running tests.

A simple way to do this is have the plugin directory as part of the knet
handle then we can override it in the test suite, which already does
'illegal' accesses to some handle fields.

OK, it's not the MOST elegant way of doing this perhaps, but it is
simple and effective.

fabbione commented:
@chrissie-c I left you a bunch of comments on IRC related to this patch. Did you have a chance to look at them?

chrissie-c committed:
[lib] Don't use -rpath

rpath seems, understandably, to annoy some analysis tools. We
use it to locate the crypto & compression plugins but libknet also needs
to use the plugins from the build-tree when running tests.

A simple way to do this is have the plugin directory as part of the knet
handle then we can override it in the test suite, which already does
'illegal' accesses to some handle fields.

OK, it's not the MOST elegant way of doing this perhaps, but it is
simple and effective.

chrissie-c commented:
Squashed.

fabbione commented:
patch looks good, thanks for the major work.

will queue for stable once CI is done (and CI itself is stable...)

fabbione commented:
@chrissie-c I force pushed a one line change to avoid a memory leak in test-common.c: free(namelist) that was failing CI.

fabbione commented:
retest this please
(ipv6 network failure on some builders)
